tests/testament/tester.nim

```diff
@@ -212,20 +212,31 @@ proc cmpMsgs(r: var TResults, expected, given: TSpec, test: TTest) =
 proc generatedFile(path, name: string, target: TTarget): string =
   let ext = targetToExt[target]
   result = path / "nimcache" /
-    (if target == targetJS: path.splitPath.tail & "_" else: "compiler_") &
+    (if target == targetJS: "" else: "compiler_") &
```

Comment:
`generatedFile` is used only when the specs needs to perform a check on the generated code. There wasn't a JS test using this until now and the code was just not correct.